### PR TITLE
Add socket mode compatibility

### PR DESCRIPTION
**Pull request by alchem0x2A**
Nov 18, 2024
Name: Tian Tian, Lucas Timmerman
Changes: (initialization.c, initialization.h, electronicGroundState.c,
	include/isddft.h, main.c, main_socket.c, makefile, md.c, relax.c,
	socket/*.h, socket/*.c, tests/Socket/*, CI workflow, doc)
1. Add socket communication layer and socket submodule
2. Unifying MLFF and DFT single point calculations to Calculate_Properties function
3. Update CI workflow for compiling socket code
4. Update socket mode simple tests
5. Update documentation for socket mode

### CODE REVIEW

**Comment by alchem0x2A** [comment]
@ltimmerman3 please review especially if all your changes to the MLFF codes are incorporated.

File: src/electronicGroundState.c [hunk 1]
```diff
@@ -62,6 +63,16 @@
 #define min(x,y) ((x)<(y)?(x):(y))
 #define TEMP_TOL 1e-12
 
+/**
+ * @brief   Calculate properties for fixed atom positions using DFT or MLFF
+ */
+void Calculate_Properties(SPARC_OBJ *pSPARC) {
```
Comment (ltimmerman3) [inline src/electronicGroundState.c:69]: Tagging comment. Need to ad declaration in include/electronicGroundState.h

**Review by ltimmerman3 — CHANGES_REQUESTED** [review]
Seems mostly okay. Made a suggestion regarding the documentation naming/searching. Need to add declaration for Calculate_Properties to include/electronicGroundState.h

File: src/electronicGroundState.c (hunk 1 above)
Comment (alchem0x2A) [reply src/electronicGroundState.c:69]: added in 4c52587ee9bba69da4756ce7abfc6fdb038bd35d

**Pull request merged**